oak/include/oak/server.hpp

```diff
@@ -173,20 +162,62 @@ class Server {
   /**
    * Add a handler with the route muxer in the internal handler routine.
    */
-  void add_handler(const std::string& key, Handler h) { handler_.add(key, h); }
+  inline void add_handler(const std::string& key, Handler h) { muxer_.add(key, h); }
 
  private:
-  // Configuration
+  void init();
+  /**
+   * @brief Asynchronously accepts connections via the socket utilizing the acceptor
+   * @param acceptor Acceptor used for receiving connections
+   * @param socket Socket used for communication
+   */
+  void serve(boost::asio::ip::tcp::acceptor& acceptor, boost::asio::ip::tcp::socket& socket);
+  /**
+   * @brief Serves/Pumps the data
+   * @note This method blocks until the server is stopped
+   */
+  void server_thread();
+  /**
+   * @brief Actual serving/pumping method used by server_thread & worker-threads
+   */
+  void server_thread_impl();
+
+  /**
+   * @brief IP-Address on which the server is listening (e.g. 0.0.0.0, ::1, 127.0.0.1)
+   */
   std::string listen_addr_;
-  int listen_port_;
-  int listen_threads_;
-
-  // State
-  bool listening_;
-  ServerImplHandler handler_;
-  std::unique_ptr<ServerImpl::options> options_;
-  std::unique_ptr<ServerImpl> server_;
-  std::unique_ptr<boost::thread> thread_;
+  /**
+   * @brief TCP Port on which the server is listening
+   */
+  unsigned short listen_port_;
+  /**
+   * @brief
+   */
+  unsigned int listen_threads_;
+  /**
+   * @brief State of the server
+   */
+  std::atomic<ServerState> state_;
+  /**
+   * @brief Cloe endpoint multiplexer
+   */
+  Muxer<Handler> muxer_;
+  /**
+   * @brief Asio IOContext
+   */
+  boost::asio::io_context ioc_;
```

Comment:
You know what they say about comments, right? I find most of these not so helpful. Better to omit them so that when we /do/ use comments, they pop! :-)